io.openems.edge.bridge.modbus/src/io/openems/edge/bridge/modbus/sunspec/AbstractOpenemsSunSpecComponent.java

```diff
@@ -348,8 +356,37 @@ protected void addBlock(int startAddress, SunSpecModel model, Priority priority)
 			}
 		}
 
-		final Task readTask = new FC3ReadRegistersTask(elements[0].startAddress, priority, elements);
-		this.modbusProtocol.addTask(readTask);
+		this.addReadTasks(elements, priority);
+	}
+
+	/**
+	 * Splits the task if it is too long and adds the read tasks.
+	 * 
+	 * @param elements	the Deque of {@link ModbusElement}s for one block.
+	 * @param priority	the reading priority
+	 * @throws OpenemsException on error
+	 */
+	private void addReadTasks(Deque<ModbusElement> elements, Priority priority) throws OpenemsException {
+		var length = 0;
+		var taskElements = new ArrayDeque<ModbusElement>();
+		var element = elements.pollFirst();
+		while (element != null) {
+			if (length + element.length > MAXIMUM_TASK_LENGTH) {
+				this.modbusProtocol.addTask(//
```

Comment:
I would have personally extracted this to another method, as it repeats 4 rather complicated lines of code right now. But it serves the job. 👍

To create an array from a list (or stream), you can also simply use `taskElements.toArray(AbstractModbusElement[]::new);`

Reply:
Okay, thanks. We can refactor that later on.